x/merkledb/trieview.go

```diff
@@ -869,16 +867,26 @@ func (t *trieView) insert(
 	// have the existing path node and the value being inserted as children.
 
 	// generate the new branch node
+	// find how many tokens are common between the existing child's compressed path and
+	// the current key(offset by the closest node's key),
+	// then move all the common tokens into the branch node
+	commonPrefixLength := getLengthOfCommonPrefix(existingChildEntry.compressedPath, key, closestNodeKeyLength+1)
+	branchKeyLength := closestNodeKeyLength + 1 + commonPrefixLength
+
+	// The existing child's key is of length: len(closestNodeKey) + 1 (for the child index) + len(existing child's compressed key)
+	// if that length is less than or equal to the branch node's key that implies that the existing child's key matched the key to be inserted.
+	// Since it matched the key to be inserted, it should have been the last node returned by GetPathTo
+	if closestNodeKeyLength+1+existingChildEntry.compressedPath.length <= branchKeyLength {
+		return nil, ErrGetPathToFailure
+	}
+
 	branchNode := newNode(
 		closestNode,
-		key[:closestNodeKeyLength+1+getLengthOfCommonPrefix(existingChildEntry.compressedPath, remainingKey)],
+		key.Take(branchKeyLength),
 	)
-	if err := t.recordNodeChange(closestNode); err != nil {
```

Comment:
why is this no longer needed?

Reply:
This was never needed once the loop on [833](https://github.com/ava-labs/avalanchego/pull/2010/files/e83a9031b93224784f24e04887e9968ec7c8b5c2#diff-c2d02688e7433bd4deb94e0ac43018c9d0be883d8183e4c21ebc2687f044d8d7L833) was added.